source/file.c

```diff
@@ -73,14 +73,25 @@ int aws_byte_buf_init_from_file(struct aws_byte_buf *out_buf, struct aws_allocat
     size_t read = fread(out_buf->buffer, 1, out_buf->len, fp);
     if (read < out_buf->len) {
         int errno_value = ferror(fp) ? errno : 0; /* Always cache errno before potential side-effect */
-        aws_translate_and_raise_io_error_or(errno_value, AWS_ERROR_FILE_READ_FAILURE);
-        AWS_LOGF_ERROR(
-            AWS_LS_COMMON_IO,
-            "static: Failed reading file:'%s' errno:%d aws-error:%s",
-            filename,
-            errno_value,
-            aws_error_name(aws_last_error()));
-        goto error;
+        if (errno_value != 0) {
+            aws_translate_and_raise_io_error_or(errno_value, AWS_ERROR_FILE_READ_FAILURE);
+            AWS_LOGF_ERROR(
+                AWS_LS_COMMON_IO,
+                "static: Failed reading file:'%s' errno:%d aws-error:%s",
+                filename,
+                errno_value,
+                aws_error_name(aws_last_error()));
+            goto error;
+        } else {
+            AWS_LOGF_WARN(
+                AWS_LS_COMMON_IO,
+                "static: reading file:'%s' reports longer length %d than actually reading from it: %d. This is not "
+                "necessarily an error as devices will usually report a page for length regardless of actual length.",
+                filename,
+                (int)read,
+                (int)out_buf->len);
```

Comment:
debatable: I realize this is only WARN, but If it's going to print every single time someone runs an app and everything is functioning normally... that's annoying. I know we have ERRORs in our codebase that print every time under normal operation and it's annoying.

But also, this might be worth warning about!

How about this as a compromise: Call fread() until EOF and just use the file_size query as a suggestion. fread() doesn't promise to give you everything you ask for in one call anyway (in theory). Also, that would let us get rid of the [alternate implementation of this function in aws-c-iot](https://github.com/awslabs/aws-c-iot/blame/9e1aae59a41fc6574cf5b093a03159e09697cc73/source/linux/network.c#L122).

Reply:
I'll do the finicky rewrite to save you time

Reply:
I rewrote it.

Reply:
Sorry, I'd already done it before I noticed your comment:
https://github.com/awslabs/aws-c-common/pull/1064

Reply:
Mine handles some more edge cases: like the OS reporting 0 file size (this actually happens), or the edge case where the buffer perfectly fills up and there's no room for the null terminator